build_libtensor.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import os
+import shutil
+import argparse
+import subprocess
+
+from os.path import join
+
+try:
+    import multiprocessing
+
+    NCPU = multiprocessing.cpu_count()
+except ImportError:
+    NCPU = 1
+
+
+def is_ninja_available():
+    status, _ = subprocess.getstatusoutput("ninja")
+    return status in [0, 1]
+
+
+def configure(build_dir, source_dir, install_dir, build_type=None, features=[],
+              use_ninja=is_ninja_available()):
+    args = ["cmake", "-DCMAKE_INSTALL_PREFIX=" + install_dir]
+
+    if use_ninja:
+        args += ["-G", "Ninja"]
+
+    if build_type in ["Release", "Debug", "MinSizeRel", "RelWithDebInfo"]:
+        args += ["-DCMAKE_BUILD_TYPE=" + build_type]
+    elif build_type == "SanitizeAddress":
+        cpflags = "-O1 -g -fsanitize=address -fno-omit-frame-pointer"
+        ldflags = "-fsanitize=address"
+        args += ["-DCMAKE_CXX_FLAGS_DEBUG={}".format(cpflags),
+                 "-DCMAKE_C_FLAGS_DEBUG={}".format(cpflags),
+                 "-DCMAKE_EXE_LINKER_FLAGS_DEBUG=\"{}\"".format(ldflags),
+                 "-DCMAKE_BUILD_TYPE=Debug"]
+    elif build_type:
+        raise SystemExit("Unknown build type: " + build_type)
+
+    if "mpi" in features:
+        args += ["-DWITH_MPI=ON"]
+    if "libxc" in features:
```

Comment:
I know you do DFT now... but this needs to be libxm 😁

Reply:
Getting tired you know ;).